### PR TITLE
intel-compute-runtime: update to 24.45.31740.9

### PR DESCRIPTION
**Pull request by ouankou**
Topic Description
-----------------

- opencl-registry-api: downgrade to 2024.05.08
    API changes in version 2024.10.24 break intel-compute-runtime.
    The version must stay at 2024.05.08 until intel-compute-runtime supports the new API.
- intel-compute-runtime: update to 24.45.31740.9
- intel-graphics-compiler: update to 2.1.12
- level-zero: update to 1.19.2
- igsc: update to 0.9.5
- intel-gmmlib: update to 22.5.4

Package(s) Affected
-------------------

- igsc: 0.9.5
- intel-compute-runtime: 24.45.31740.9
- intel-gmmlib: 22.5.4
- intel-graphics-compiler: 2.1.12
- level-zero: 1.19.2
- opencl-registry-api: 1:2024.05.08

Security Update?
----------------

No

Build Order
-----------

```
#buildit opencl-registry-api intel-gmmlib igsc level-zero intel-graphics-compiler intel-compute-runtime
```

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by ouankou** [comment]
Tested on i9-11950H.
```bash
(ssh)ouankou@ei [ ~ ] $ clpeak -p 3

Platform: Intel(R) OpenCL Graphics
  Device: Intel(R) UHD Graphics
    Driver version  : 24.45.31740 (Linux x64)
    Compute units   : 32
    Clock frequency : 1450 MHz

    Global memory bandwidth (GBPS)
      float   : 42.43
      float2  : 42.82
      float4  : 42.48
      float8  : 42.41
      float16 : 32.04

    Single-precision compute (GFLOPS)
      float   : 719.50
      float2  : 711.00
      float4  : 717.66
      float8  : 455.26
      float16 : 451.99

    Half-precision compute (GFLOPS)
      half   : 1392.94
      half2  : 1385.33
      half4  : 1392.44
      half8  : 1389.04
      half16 : 1264.90

    No double precision support! Skipped

    Integer compute (GIOPS)
      int   : 242.63
      int2  : 163.85
      int4  : 156.39
      int8  : 148.71
      int16 : 180.50

    Integer compute Fast 24bit (GIOPS)
      int   : 242.31
      int2  : 163.85
      int4  : 156.54
      int8  : 148.71
      int16 : 180.49

    Transfer bandwidth (GBPS)
      enqueueWriteBuffer              : 20.73
      enqueueReadBuffer               : 20.73
      enqueueWriteBuffer non-blocking : 21.25
      enqueueReadBuffer non-blocking  : 21.41
      enqueueMapBuffer(for read)      : 1717983.62
        memcpy from mapped ptr        : 20.75
      enqueueUnmap(after write)       : 2147479.50
        memcpy to mapped ptr          : 20.72

    Kernel launch latency : 26.16 us
```

**Comment by ouankou** [comment]
The upgrade also resolved the issue: https://github.com/AOSC-Dev/aosc-os-abbs/issues/7879. Flag `-g` works again.
```bash
(ssh)ouankou@ei [ reduction@main ] $ icx -fiopenmp -fopenmp-targets=spir64 test.c -O0 -g
(ssh)ouankou@ei [ reduction@main ] $ LIBOMPTARGET_INFO=1 ./a.out
omptarget device 0 info: Entering OpenMP kernel at test.c:0:14 with 7 arguments:
omptarget device 0 info: tofrom(sum)[4]
omptarget device 0 info: private(unknown)[4096]
omptarget device 0 info: private(unknown)[4]
omptarget device 0 info: private(unknown)[4]
omptarget device 0 info: firstprivate(cmplr.tmp.3 (compiler-generated normalized.lb for i-loop))[0]
omptarget device 0 info: firstprivate(cmplr.tmp.2 (compiler-generated normalized.ub for i-loop))[0]
omptarget device 0 info: alloc(unknown)[32]
sum = 15050
omptarget device 0 info: Entering OpenMP kernel at test.c:24:24 with 0 arguments:
```

File: app-devel/intel-graphics-compiler/autobuild/prepare [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-export CXXFLAGS="${CXXFLAGS} -fno-semantic-interposition -I "${SRCDIR}"/../SPIRV-LLVM-Translator/include"
+export CXXFLAGS="${CXXFLAGS} -fno-semantic-interposition"
 export LDFLAGS="${LDFLAGS} -Wl,-Bsymbolic"
 
 git config --local user.name "AOSC Maintainers"
 git config --local user.email "maintainers@aosc.io"
 
-LLVM_VER=15.0.7
+LLVM_VER=14.0.6
 mv "${SRCDIR}"/../llvm-project-"${LLVM_VER}".src "${SRCDIR}"/../llvm-project
```
Comment (OriginCode) [inline app-devel/intel-graphics-compiler/autobuild/prepare]: Add some verbosity.

File: app-devel/intel-graphics-compiler/autobuild/prepare [hunk 2]
```diff
@@ -15,12 +15,12 @@ do
     patch -p1 -t -s -N -i "$file"
 done
 rm "${SRCDIR}"/../opencl-clang/patches/clang/*.patch
```
Comment (OriginCode) [inline app-devel/intel-graphics-compiler/autobuild/prepare]: Same as above.

File: app-devel/intel-graphics-compiler/autobuild/prepare [hunk 3]
```diff
@@ -15,12 +15,12 @@ do
...
-for file in "${SRCDIR}"/external/llvm/releases/15.0.0/patches_external/*.patch
+for file in "${SRCDIR}"/external/llvm/releases/14.0.0/patches_external/*.patch
 do
     patch -p1 -t -s -N -i "$file"
 done
-rm "${SRCDIR}"/external/llvm/releases/15.0.0/patches_external/*.patch
+rm "${SRCDIR}"/external/llvm/releases/14.0.0/patches_external/*.patch
```
Comment (OriginCode) [inline app-devel/intel-graphics-compiler/autobuild/prepare]: Same here.

File: app-devel/intel-graphics-compiler/autobuild/prepare [hunk 4]
```diff
@@ -15,12 +15,12 @@ do
...
 
 mkdir "${SRCDIR}"/../llvm-project/.git
-ln -s "${SRCDIR}"/../SPIRV-LLVM-Translator  "${SRCDIR}"/../llvm-project/llvm/projects/llvm-spirv
+ln -s "${SRCDIR}"/../SPIRV-LLVM-Translator-IGC "${SRCDIR}"/../llvm-project/llvm/projects/llvm-spirv
 ln -s "${SRCDIR}"/../opencl-clang "${SRCDIR}"/../llvm-project/llvm/projects/opencl-clang
```
Comment (OriginCode) [inline app-devel/intel-graphics-compiler/autobuild/prepare]: Same here.

**Review by OriginCode — DISMISSED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**